openslides/agenda/static/js/agenda/agenda.js

```diff
@@ -190,199 +201,270 @@ angular.module('OpenSlidesApp.agenda.site', ['OpenSlidesApp.agenda'])
-    Agenda.bindOne(item.id, $scope, 'item');
-    User.bindAll({}, $scope, 'users');
-    $scope.speaker = {};
-    $scope.alert = {};
-
-    // close/open list of speakers of current item
-    $scope.closeList = function (listClosed) {
-        item.speaker_list_closed = listClosed;
-        Agenda.save(item);
-    };
-    // add user to list of speakers
-    $scope.addSpeaker = function (userId) {
-        $http.post('/rest/agenda/item/' + item.id + '/manage_speaker/', {'user': userId})
-            .success(function(data){
-                $scope.alert.show = false;
-            })
-            .error(function(data){
-                $scope.alert = { type: 'danger', msg: data.detail, show: true };
-            });
-    };
-    // delete speaker(!) from list of speakers
-    $scope.removeSpeaker = function (speakerId) {
-        $http.delete('/rest/agenda/item/' + item.id + '/manage_speaker/',
-                {headers: {'Content-Type': 'application/json'},
-                 data: JSON.stringify({speaker: speakerId})})
-            .error(function(data){
-                $scope.alert = { type: 'danger', msg: data.detail, show: true };
-            });
-    };
-    // begin speech of selected/next speaker
-    $scope.beginSpeech = function (speakerId) {
-        $http.put('/rest/agenda/item/' + item.id + '/speak/', {'speaker': speakerId})
-            .success(function(data){
-                $scope.alert.show = false;
-            })
-            .error(function(data){
-                $scope.alert = { type: 'danger', msg: data.detail, show: true };
-            });
-    };
-    // end speech of current speaker
-    $scope.endSpeech = function () {
-        $http.delete('/rest/agenda/item/' + item.id + '/speak/',
-                {headers: {'Content-Type': 'application/json'},
-                 data: JSON.stringify()})
-            .error(function(data){
-                $scope.alert = { type: 'danger', msg: data.detail, show: true };
+.controller('ItemListCtrl', [
+    '$scope',
+    '$http',
+    '$state',
+    'Agenda',
+    'AgendaTree',
+    'Projector',
+    function($scope, $http, $state, Agenda, AgendaTree, Projector) {
+        // Bind agenda tree to the scope
+        $scope.$watch(function () {
+            return Agenda.lastModified();
+        }, function () {
+            $scope.items = AgendaTree.getFlatTree(Agenda.getAll());
+        });
+
+        // open detail view link
+        $scope.openDetail = function (id) {
+            $state.go('agenda.item.detail', {id: id});
+        }
+
+        // save changed item
+        $scope.save = function (item) {
+            Agenda.save(item);
+        };
+
+        // *** delete mode functions ***
+        $scope.isDeleteMode = false;
+        // check all checkboxes
+        $scope.checkAll = function () {
+            angular.forEach($scope.items, function (node) {
+                node.selected = $scope.selectedAll;
```

Comment:
Could you rename 'node' to 'item'? The name 'node' comes from the time, where the tree was seperate from the items.